webhook/batch_url_notifier.go

```diff
@@ -0,0 +1,140 @@
+package webhook
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"github.com/livekit/protocol/auth"
+	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/logger"
+	"google.golang.org/protobuf/encoding/protojson"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type BatchURLNotifierParams struct {
+	Logger    logger.Logger
+	URL       string
+	Interval  time.Duration
+	MaxSize   int
+	APIKey    string
+	APISecret string
+}
+
+type BatchURLNotifier struct {
+	client  *http.Client
+	mu      sync.RWMutex
+	params  BatchURLNotifierParams
+	batch   []*livekit.WebhookEvent
+	dropped int // it's operated inside a mutex scope so no need for atomic type
+	done    chan struct{}
+}
+
+func NewBatchURLNotifier(params BatchURLNotifierParams) URLNotifier {
+	if params.Interval == 0 {
+		params.Interval = 100 * time.Millisecond
```

Comment:
add default constants for `params.Interval` and `params.MaxSize` at the top of this file.

Reply:
done